zc_events/client.py

```diff
@@ -158,23 +158,22 @@ def handle_request_event(self, event, view=None, viewset=None, relationship_view
         relationship = event.get('relationship', None)
         related_resource = event.get('related_resource', None)
 
+        handler_kwargs = {}
         if view:
             handler = view.as_view()
-        elif pk and relationship:
-            handler = relationship_viewset.as_view()
-        elif request.method == 'GET' and pk and related_resource:
-            handler = viewset.as_view({'get': related_resource})
+        elif pk:
+            handler_kwargs['pk'] = pk
+            if relationship:
+                handler_kwargs['related_field'] = relationship
```

Comment:
Here's where we convert `relationship` to `related_field`, just before we call the view.